lib/fluent/plugin/kubernetes_metadata_watch_pods.rb

```diff
@@ -70,6 +86,14 @@ def start_pod_watch
             @stats.bump(:pod_cache_watch_ignored)
         end
       end
+    rescue Exception => e
+      # Instead of raising exceptions and crashing Fluentd, swallow the
+      # exception and reset the watcher.
+      log.info "Exception encountered parsing pod watch event. " \
+               "The connection might have been closed. " \
+               "Resetting the pod watcher.", e
+      Thread.current[:watcher_alive] = false
```

Comment:
Why is this written through `Thread.current[:watcher_alive]`, but read with `@watcher_alive` above?

Reply:
Fixed.